test/functional/wallet_labels.py

```diff
@@ -50,6 +50,27 @@ def invalid_label_name_test(self):
         for rpc_call in rpc_calls:
             assert_raises_rpc_error(-11, "Invalid label name", *rpc_call, label="*")
 
+    def test_label_named_parameter_handling(self):
+        """Test that getnewaddress with labels containing '=' characters is handled correctly in -named mode"""
+        self.log.info("Test getnewaddress label parameter handling")
+        node = self.nodes[0]
+
+        # Test getnewaddress with explicit named parameter containing '='
+        label_with_equals = "wallet=wallet"
+        result = node.cli("-named", "getnewaddress", f"label={label_with_equals}").send_cli()
+        address = result.strip()
+        addr_info = node.getaddressinfo(address)
+        assert_equal(addr_info.get('labels', []), [label_with_equals])
+
+        self.log.info("Test bitcoin-cli -named passes parameter containing '=' by position if it does not specify a known parameter name and is in a string position")
+        equals_label = "my=label"
+        result = node.cli("-named", "getnewaddress", equals_label).send_cli()
+        address = result.strip()
+        addr_info = node.getaddressinfo(address)
+        assert_equal(addr_info.get('labels', []), [equals_label])
+
+        self.log.info("getnewaddress label parameter handling test completed successfully")
```

Comment:
In f53dbbc5057b6f676db4be9bc720898149f293fc "test: Add functional tests for named argument parsing"

nit: Logging at the end of the test case is unnecessary